BRCA_Deep_Dive.Rmd

```diff
@@ -0,0 +1,1511 @@
+       aes(x = year, y = runoff_in_wbm, color = CF, fill = CF)) + 
+   geom_line(lwd = 1) +
+  theme_bw() +
+  scale_color_manual("",values = c("black","red3","dodgerblue2")) +
+  scale_fill_manual("",values = c("black","red3","dodgerblue2")) +
+  labs(y = "Runoff (in/y)", x = "")
+
+
+# Examply daily plot
+ggplot(source_wbm %>% dplyr::filter(year(date) > 2010, year(date) < 2012), 
+       aes(x = date, y = runoff_in_wbm, color = CF, fill = CF)) + 
+   geom_line(lwd = 1) +
+  theme_bw() +
+  scale_color_manual("",values = c("black","red3","dodgerblue2")) +
+  scale_fill_manual("",values = c("black","red3","dodgerblue2")) +
+  labs(y = "Runoff (in/d)", x = "", 
+       title = "Example year when GridMET overlaps MACA. \nNote how monsoon runoff increases while spring runoff decreases\n for Warm Wet future.")
+
+```
+
+
+
+###10. WBM Runoff Routing
+
+#### 10.1 Calibrate to SW
+
+Use routing function to calibrate to discharge at nearest NWIS gage.
+
+```{r nwis_wbm}
+
+# First run WBM 
+
+# Here's an example with new "Direct_Frac" and "Return_Rate" routing parameters.
+nwis_wbm <- run_NPS_wbm(points = nwis_pts,
+                        climate_data = clim_nwis,
+                        col_name = "site_no",
+                        hock_coef = 4,
+                        PET_mult = 1,
+                        Soil_mult = 5,
+                        Direct_Frac = 0.15, 
+                        Return_Rate = 0.02) %>%
+  left_join(., nwis_nearest_discharge, by = c("date","site_no")) %>%
+  ungroup()
+
+ggplot(nwis_wbm %>% dplyr::filter(year(date) > 2010, year(date) < 2013)) + 
+  geom_line(aes(x = date, y = excess_water_in_wbm, color = "Old Runoff")) +
+  geom_line(aes(x = date, y = runoff_in_wbm, color = "New Runoff")) + 
+  geom_line(aes(x = date, y = discharge_in_nwis, color = "NWIS Discharge")) +
+  scale_color_manual("",values = c("black","tomato","dodgerblue")) +
+  theme_bw()
+
+
+
+
+# Here's an example with subsequent routing function with calibration and
+# separating out test and validation datasets
+
+# First, run wbm without Direct_Frac and Return_Rate effects
+nwis_wbm <- run_NPS_wbm(points = nwis_pts,
+                        climate_data = clim_nwis,
+                        col_name = "site_no",
+                        hock_coef = 4,
+                        PET_mult = 1,
+                        Soil_mult = 5,
+                        Direct_Frac = 0, 
+                        Return_Rate = 1) %>%
+  left_join(., nwis_nearest_discharge, by = c("date","site_no")) %>%
+  ungroup()
+
+# Define test dataset (on which to calibrate)
+test_dates <- interval(as.Date("1995-01-01"), as.Date("2005-01-01"))
+test_dat <- nwis_wbm %>%
+  dplyr::filter(date %within% test_dates)
+
+# And validation dataset (data not used for calibration)
+val_dates <- interval(as.Date("2005-01-01"), as.Date("2015-01-01"))
+val_dat <- nwis_wbm %>%
+  dplyr::filter(date %within% val_dates)
+
+# Now, calibrate...
+
+# First define fitting function for calibration
```

Comment:
Workflow Q, do we want all functions to be external .R scripts? I vote yes...I also know this is just a draft workflow though